components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -254,8 +259,44 @@ class GeckoEngineSession(
         }
 
         if (reload) {
-            this.reload()
+            if (overrideUrl == null) {
+                this.reload()
+            } else {
+                loadUrl(overrideUrl, flags = LoadUrlFlags.select(LoadUrlFlags.LOAD_FLAGS_REPLACE_HISTORY))
+            }
+        }
+    }
+
+    /**
+     * Checks and returns a non-mobile version of the url.
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun checkForMobileSite(url: String): String? {
+        var overrideUrl: String? = null
+        val mPrefix = "m."
+        val mobilePrefix = "mobile."
+
+        val uri = URI(url)
+        val authority = uri.authority?.toLowerCase(Locale.ROOT) ?: return null
+
+        val foundPrefix = when {
+            authority.startsWith(mPrefix) -> mPrefix
+            authority.startsWith(mobilePrefix) -> mobilePrefix
+            else -> null
         }
+
+        foundPrefix?.let {
+            val mobileUri = URI(
+                uri.scheme,
+                authority.substring(it.length),
+                uri.path,
+                uri.query,
+                uri.fragment
+            )
+            overrideUrl = mobileUri.toString()
```

Comment:
Could just return here and avoid the `var`

Reply:
detekt `ReturnCount = 2` - https://arturbosch.github.io/detekt/style.html. Having said that, there may be better way to do this without using var 🤔 .

Reply:
If detekt is complaining about return count then `var` is fine